### PR TITLE
[coretext] Add hb_coretext_font_create()

### PR DESCRIPTION
**Pull request by behdad**
Uncompiled, untested.

Fixes https://github.com/harfbuzz/harfbuzz/issues/628

New API:
hb_coretext_font_create()

### CODE REVIEW

File: src/hb-coretext.cc [hunk 1]
```diff
@@ -306,6 +299,35 @@ _hb_coretext_shaper_font_data_destroy (hb_coretext_shaper_font_data_t *data)
...
+  CGFontRef cg_font = CTFontCopyGraphicsFont (ct_font, 0);
+  hb_face_t *face = hb_coretext_face_create (cg_font);
+  CFRelease (cg_font);
+  hb_font_t *font = hb_font_create (face);
+  hb_face_destroy (face);
+
+  if (unlikely (hb_object_is_inert (font)))
+    return font;
+
+  /* Let there be dragons here... */
```
Comment (drott) [inline src/hb-coretext.cc]: ?

Comment (behdad) [reply src/hb-coretext.cc]: Normally that value (ctfont) is automatically created on-demand by HarfBuzz generic layer.  We are sneaking in and setting it here.  It works and is correct, but is fragile if the rest of the code changes....

**Comment by behdad** [comment]
@ebraminio since when we started -Werror'ing? So, Mac build has been broken since, because of the deprecation warnings?

**Comment by behdad** [comment]
Ok I see I'm the one who added that...

**Comment by behdad** [comment]
Can we add -Wno-deprecated-declarations to OS X builds?

**Comment by behdad** [comment]
https://github.com/harfbuzz/harfbuzz/issues/345

**Comment by ebraminio** [comment]
> Can we add -Wno-deprecated-declarations to OS X builds?

Does having access to a macOS terminal help you perhaps to try that or #486? If so, I guess CircleCI provides macOS machines with SSH access.

**Comment by behdad** [comment]
> Can we add -Wno-deprecated-declarations to OS X builds?
> 
> Does having access to a macOS terminal help you perhaps to try that or #486? If so, I guess CircleCI provides macOS machines with SSH access.

Seems like I fixed it.  ssh to mac would be nice if it works.  But if it's too much work to figure out every time I needed every six months, easier to just sit next to drott physically somewhere in the world and do it together.

**Comment by ebraminio** [comment]
Turned out CircleCI's macOS support needs some fee unlike TravisCI :(

**Pull request merged**